### ISSUE TITLE
INTERNAL Error: Attempted to dereference unique_ptr that is NULL!

### ISSUE THREAD

**Issue by rleoneti** [duckdb/duckdb#9005]
### What happens?

Query CSV with wildcard and filter for specific filename, crashes DuckDB

```
D SELECT * FROM read_csv_auto('teste*.csv',header=TRUE,filename=true,union_by_name=True);
┌───────┬─────────┬────────┬────────────┐
│  id   │  name   │ factor │  filename  │
│ int64 │ varchar │ double │  varchar   │
├───────┼─────────┼────────┼────────────┤
│     1 │ Ricardo │    1.5 │ teste1.csv │
│     2 │ Jose    │    2.0 │ teste1.csv │
│     3 │ Maria   │    2.5 │ teste2.csv │
│     4 │ Ana     │    3.0 │ teste2.csv │
└───────┴─────────┴────────┴────────────┘
D SELECT * FROM read_csv_auto('teste*.csv',header=TRUE,filename=true,union_by_name=True) where filename='teste1.csv';
Error: INTERNAL Error: Attempted to dereference unique_ptr that is NULL!
D
```

### To Reproduce

SELECT * FROM read_csv_auto('teste*.csv',header=TRUE,filename=true,union_by_name=True) where filename='teste1.csv';

### OS:

Windows x64

### DuckDB Version:

v0.8.1 6536a77232

### DuckDB Client:

Command Line

### Full Name:

Ricardo Leoneti

### Affiliation:

Data Developer

### Have you tried this on the latest `main` branch?

I have tested with a release build (and could not test with a main build)

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

**Comment by pdet** [comment]
Hi @rleoneti, many thanks for reporting the issue and providing a minimal reproducible example! That is extremely valuable!

I've pushed a fix on #9013. Could you give that a go?

**Comment by jivanic-demystdata** [comment]
Same error here: https://github.com/duckdb/duckdb-java/issues/137, probably not same issue tho
